Lib/ufo2ft/featureWriters/kernFeatureWriter.py

```diff
@@ -32,62 +47,229 @@ def unicodeBidiType(uv):
+        # self as is.
+        if (
+            len(allFirstScripts.keys()) == 1
+            and allFirstScripts.keys() == allSecondScripts.keys()
+        ):
+            for script in list(allFirstScripts.keys())[0]:
+                yield script, self
+            return
+
+        # Now let's go through the script combinations
+        for firstScripts, secondScripts in itertools.product(
+            allFirstScripts.keys(), allSecondScripts.keys()
+        ):
+            localPair = KerningPair(
+                sorted(allFirstScripts[firstScripts]),
+                sorted(allSecondScripts[secondScripts]),
+                self.value,
+                scripts=self.scripts,
+                directions=self.directions,
+                bidiTypes=self.bidiTypes,
+            )
+            # Handle very obvious common cases: one script, same on both sides
+            if (
+                len(firstScripts) == 1
+                and len(secondScripts) == 1
+                and firstScripts == secondScripts
+            ):
+                localPair.scripts = {firstScripts[0]}
+                yield firstScripts[0], localPair
+            # First is single script, second is common
+            elif len(firstScripts) == 1 and set(secondScripts).issubset(DFLT_SCRIPTS):
+                localPair.scripts = {firstScripts[0]}
+                yield firstScripts[0], localPair
+            # First is common, second is single script
+            elif set(firstScripts).issubset(DFLT_SCRIPTS) and len(secondScripts) == 1:
+                localPair.scripts = {secondScripts[0]}
+                yield secondScripts[0], localPair
+            # One script and it's different on both sides and it's not common
+            elif len(firstScripts) == 1 and len(secondScripts) == 1:
+                logger = ".".join([self.__class__.__module__, self.__class__.__name__])
+                logging.getLogger(logger).info(
+                    "Mixed script kerning pair %s ignored" % localPair
+                )
+                pass
+            else:
+                # At this point, we have a pair which has different sets of
+                # scripts on each side, and we have to find commonalities.
+                # For example, the pair
+                #   [A A-cy] {Latn, Cyrl}  --  [T Te-cy Tau] {Latn, Cyrl, Grek}
+                # must be split into
+                #   A -- T
+                #   A-cy -- Te-cy
+                # and the Tau ignored.
+                commonScripts = set(firstScripts) & set(secondScripts)
+                commonFirstGlyphs = set()
+                commonSecondGlyphs = set()
+                for scripts, glyphs in allFirstScripts.items():
+                    if commonScripts.issubset(set(scripts)):
+                        commonFirstGlyphs |= set(glyphs)
+                for scripts, glyphs in allSecondScripts.items():
+                    if commonScripts.issubset(set(scripts)):
```

Comment:
These `issubset` calls don't catch `scripts` being `Zyyy`, i.e. Common, which they should I think? 🤔